### PR TITLE
Embed search and selectionPane into rootLayout

### PR DESCRIPTION
**Pull request by justinvdk**
Make ctrl+f request focus and esc un-focus search
Start refactor of SelectionManager
Make returned node more flexible
TreeLeaf- and TreeNodeDescription now return a more flexible javaFX Node
Change graphNode description pane FXML
Removed unreliable test
Make use of fx:include for the root layout
Make FXMLLoader usage somewhat easier to read
Rename treeManager -> TreePaneController
This is more consistent with the member names and fxml ids
Rename treeManager -> TreePaneController
This is more consistent with the member names and fxml ids
Make graphPaneController responsible for its own nodes

In short: rootlayout now embeds search pane and selection info pane.

### CODE REVIEW

**Comment by justinvdk** [comment]
See #203. Will open PR for that when this one is merged.

**Comment by cbilstra** [comment]
From what I can see a lot of small changes, looks good.

File: PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/selection/SelectionPaneController.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    rootPane.setBackground(new Background(
+        new BackgroundFill(BACKGROUND_COLOR, null, Insets.EMPTY)));
+  }
+
+  /**
+   * Sets the content of the selectionPane.
+   *
+   * <p>
+   *   This removes previous shown content,
+   *   call with null to clear.
```
Comment (wouterjsmit) [inline PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/selection/SelectionPaneController.java]: Please remove the indents here. IntelliJ makes them automatically but shouldn't be there.

File: PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/selection/SelectionPaneController.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+   * </p>
+   *
+   * @param content new content, may be null
+   */
+  public void setContent(Node content) {
+    rootPane.getChildren().clear();
+    if (content != null) {
```
Comment (wouterjsmit) [inline PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/selection/SelectionPaneController.java:40]: This is simply fugly. The whole method's functionality is inside the not-null condition.

You could use @nullable or something to denote that it may be `null` (not sure if that's an IntelliJ or JavaDoc feature though).

Why may it be null? IMO the caller should separately use something like `clearContent()` to remove the content. Not entirely sure where this is called, though. This separates the clearing part (outside of the condition) from the fillling part (inside of the condition).

Comment (justinvdk) [inline PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/selection/SelectionPaneController.java:40]: I'll introduce a clearContent method for now. It emerged like this because setContent was already clearing the view anyway.

File: PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/selection/SelectionPaneController.java (hunk 1 above)
Comment (justinvdk) [inline PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/view/selection/SelectionPaneController.java]: Odd, I've changed this in a commit but the preview still shows the old code?

**Comment by wouterjsmit** [comment]
Changes in this PR look good. The test that you removed is unstable, but Faris also removed it. One of you will have to rebase against the other and remove his commit that tries to redo the action. In this case, probably Faris

**Comment by wouterjsmit** [comment]
Ready for Merging I think?
